src/decimal128.ts

```diff
@@ -342,10 +346,9 @@ export class Decimal128 extends BSONValue {
       // Shift exponent to significand and decrease
       lastDigit = lastDigit + 1;
 
-      if (lastDigit - firstDigit > MAX_DIGITS) {
+      if (lastDigit - firstDigit >= MAX_DIGITS) {
```

Comment:
Check this with a test case

Reply:
Looking at it, `firstDigit` is always 0; it doesn't change at all throughout the entire function, so I'm going to go ahead and remove all references to it.

With regards to changing to the gte sign, the cases that trigger this branch are clamped zeros with large positive increments (see spec tests in decimal128-1.json), but any number with a sufficiently large exponent can enter this block.

If we consider a number like `1000000000000000000000000000000000e6112`, which has 1 significant digit, 34 digits before the exponent and an exponent that's above max exponent, then what happens when we have a gt sign here is that we end up with the following result:

```
> Decimal128.fromString('1000000000000000000000000000000000e6112')
new Decimal128("0E+6111")
```

since we skip the iteration when `lastDigit` reaches `MAX_DIGITS` and proceed without error despite the fact that we should have overflowed. This leads us to get to our encoding and produce the incorrect result we see above.

With the gte sign in place, we get the following result, which is what we expect

```
> Decimal128.fromString('1000000000000000000000000000000000e6112')
Uncaught:
BSONError: "1000000000000000000000000000000000e6112" is not a valid Decimal128 string - overflow
    at invalidErr (/home/wajames/js-bson/lib/bson.cjs:1402:11)
    at Decimal128.fromString (/home/wajames/js-bson/lib/bson.cjs:1540:17)
```

it seems like we should have spec test that test corner cases like this.